3.4/graphs-pregel.md

```diff
@@ -403,3 +407,24 @@ const handle = pregel.start("slpa", "yourgraph", {maxGSS:100, resultField:"commu
 // check the status periodically for completion
 pregel.status(handle);
 ```
+
+Limits
+------
+
+Pregel algorithms in ArangoDB will by default store temporary vertex and edge data in main memory. For large
+datasets this is going to cause problems, as servers may run out of memory while loading the data. 
+
+To avoid servers from running out of memory while loading the dataset, a Pregel job can be started with the
+attribute `useMemoryMaps` set to `true`. This will make the algorithm use memory-mapped files as a backing
+storage in case of huge datasets. Falling back to memory-mapped files might make the computation disk-bound, but
+may be the only way to complete the computation at all.
+
+Parts of the Pregel temporary results (aggregated messages) may also be stored in main memory, and currently
+the aggregation cannot fall back to memory-mapped files. That means if an algorithm needs to store a lot of
+result messages temporarily, it may consume a lot of main memory.
+
+In general it is also recommended to set the `store` attribute of Pregel jobs to `true` to make a job store
+its value on disk and not just in main memory. This way the results are removed from main memory once a Pregel 
+job completes. If the `store` attribute is explicitly set to `false`, result sets of completed Pregel runs
+will not be removed from main memory unless the result set is explicitly discarded by a call to the `cancel`
```

Comment:
Is it kept forever if no one calls the cancel method? And removed from memory *when* cancel is called?

```suggestion
will not be removed from main memory until the result set is explicitly discarded by a call to the `cancel`
```